### PR TITLE
Get task info by runId

### PR DESCRIPTION
**Pull request by ssalinas**
Get the `SingularityTaskIdHistory` object for a task by it's run id

/cc @jhaber this should get you back taskId, last state, and the SingularityTaskId object

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java [hunk 1]
```diff
@@ -29,6 +29,9 @@
   @SqlQuery("SELECT bytes FROM taskHistory WHERE taskId = :taskId")
   byte[] getTaskHistoryForTask(@Bind("taskId") String taskId);
 
+  @SqlQuery("SELECT bytes FROM taskHistory WHERE runId = :runId")
+  byte[] getTaskHistoryForTaskByRunId(@Bind("runId") String runId);
```
Comment (jhaber) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java]: Shouldn't this also filter by requestId?

File: SingularityService/src/main/java/com/hubspot/singularity/resources/HistoryResource.java [hunk 2]
```diff
@@ -153,6 +153,18 @@ public SingularityDeployHistory getDeploy(@ApiParam("Request ID for deploy") @Pa
   }
 
   @GET
+  @Path("/request/{requestId}/tasks/{runId}")
```
Comment (jhaber) [inline SingularityService/src/main/java/com/hubspot/singularity/resources/HistoryResource.java]: Should this be `/request/{requestId}/run/{runId}`?

File: SingularityService/src/main/java/com/hubspot/singularity/resources/HistoryResource.java [hunk 3]
```diff
@@ -153,6 +153,18 @@ public SingularityDeployHistory getDeploy(@ApiParam("Request ID for deploy") @Pa
...
+  @ApiOperation("Retrieve the history for all tasks of a specific request.")
```
Comment (jhaber) [inline SingularityService/src/main/java/com/hubspot/singularity/resources/HistoryResource.java]: This doesn't sound right

File: SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java [hunk 4]
```diff
@@ -29,6 +29,9 @@
   @SqlQuery("SELECT bytes FROM taskHistory WHERE taskId = :taskId")
   byte[] getTaskHistoryForTask(@Bind("taskId") String taskId);
 
+  @SqlQuery("SELECT bytes FROM taskHistory WHERE requestId = :requestId AND runId = :runId")
+  byte[] getTaskHistoryForTaskByRunId(@Bind("requestId") String requestId, @Bind("runId") String runId);
```
Comment (wsorenson) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java]: do we index on runId?

**Comment by ssalinas** [comment]
Updated a couple things, had forgotten about the description, thanks for that

File: SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java (hunk 4 above)
Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java]: not currently, we would probably need to for this case

Comment (tpetr) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java]: yeah, we should add an index. let me know if you need a hand with this.

Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java]: Added an index in  b63cf47 , can someone with more mysql chops double check that please?

Comment (tpetr) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java]: syncing up w/ monty about this

Comment (solomonty) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java]: Which query/queries will benefit from the added index?

Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java]: The one on the line for this comment thread:
`SELECT bytes FROM taskHistory WHERE requestId = :requestId AND runId = :runId`

**Comment by tpetr** [comment]
This PR reads like there was an assumption that `runId` would be unique, but I don't see any constraint for that in #669. Did I miss something, or was this an oversight?

**Pull request merged**